crates/evm/evm.rs

```diff
@@ -79,6 +83,31 @@ pub fn execute_tx(
     run_evm(tx_env, block_env, state, spec_id)
 }
 
+// Executes a single GenericTransaction, doesn't commit the result or perform state transitions
+pub fn simulate_tx_from_generic(
+    tx: &GenericTransaction,
+    header: &BlockHeader,
+    state: &mut EvmState,
+    spec_id: SpecId,
+) -> Result<ExecutionResult, EvmError> {
+    let block_env = block_env(header);
+    let tx_env = tx_env_from_generic(tx, header.base_fee_per_gas);
+    run_without_commit(tx_env, block_env, state, spec_id)
+}
+
+fn adjust_base_fee(
```

Comment:
where does this come from? A comment or link to a discussion on this could be useful.